vllm/sampling_params.py

```diff
@@ -51,7 +52,7 @@ def __init__(
         top_p: float = 1.0,
         top_k: int = -1,
         use_beam_search: bool = False,
-        stop: Union[str, List[str]] = [],
+        stop: Union[str, Sequence[str]] = (),
```

Comment:
I'd like to keep this as the `List` type, because the type makes it easy to reason about the `stop` object. For example, if we need to internally add a new string to `stop`, then we can simply use `stop.append()`. Also, I believe this is a reasonable restriction that doesn't hurt UX.